packages/mp/src/mp/validate/pre_build_validation.py

```diff
@@ -70,11 +79,104 @@ def run_pre_build_validation(self) -> None:
         )
 
     def _get_validation_functions(self) -> list[Callable]:
-        return [
-            self._uv_lock_validation,
-        ]
+        return [self._uv_lock_validation, self._version_bump_validation]
 
     def _uv_lock_validation(self) -> None:
         self.results.errors.append("[yellow]Running uv lock validation [/yellow]")
         if not mp.core.file_utils.is_built(self.integration_path):
             mp.core.unix.check_lock_file(self.integration_path)
+
+    def _version_bump_validation(self) -> None:
+        self.results.errors.append("[yellow]Running version bump validation [/yellow]")
+
+        head_sha = os.environ.get("GITHUB_SHA")
+        if not head_sha:
+            return
+
+        changed_files: list[pathlib.Path] = mp.core.unix.get_changed_files_from_main(
+            "main", head_sha, self.integration_path
+        )
+        if not changed_files:
+            return
+
+        relevant_files: list[pathlib.Path] = [
+            p for p in changed_files if p.name in {"pyproject.toml", "release_notes.yaml"}
+        ]
+        if not relevant_files or len(relevant_files) != REQUIRED_CHANGED_FILES_NUM:
+            raise NonFatalValidationError(
+                ERROR_MSG.format(
+                    "project.toml and release_notes.yml files must be updated before PR"
+                )
+            )
```

Comment:
Why don't you just use
```suggestion
            msg: str = "project.toml and release_notes.yml files must be updated before PR"
            raise NonFatalValidationError(msg)
```